### PR TITLE
Restore prior windows installer option for start menu links

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------

Make create start menu links optional in QtIFW for windows.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add to input rules file for interfaces
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Required documentation updates?
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Comment by jmarrec** [comment]
@mjwitte Built fine with windows. Unselecting "Create Start Menu" will indeed not create any start menu links. You will still be prompted to enter a name for the start menu but it will not be used

**Comment by mjwitte** [comment]
@jmarrec @Myoldmopar Made a fresh clone, switched to this branch, merged in develop, ran cmake, tried to build package. Failed, never built doc pdfs. Here's the VS build output.
[Package build output.txt](https://github.com/NREL/EnergyPlus/files/3543957/Package.build.output.txt)

**Comment by jmarrec** [comment]
@mjwitte  it should work now.

**Comment by mjwitte** [comment]
Hmm, no difference here, still no pdfs generated. I'm still using VS2017, but I don't think that matters. I'm configuring cmake with build docs, fortran, testing, and package. 
Fails with this branch.
Fails with develop.
Fails with commit 5706a997b3d182600eac398f1cd82b7d8a0e894e (prior to merge of #7334). 
Works with v9.2.0-IOFreeze tag.
So, I'm confused. I don't see any relevant commits to installer stuff between v9.2.0-IOFreeze and 5706a99.

**Comment by jmarrec** [comment]
Speaking with @mjwitte it may be because he has CMake 3.11.2 and I have messed up a VERSION_GREATER statement. I wanted to say `VERSION_GREATER_EQUAL 3.12`, except version_greater_equal doesn't exist until cmake 3.7 when we apparently support 2.8... and I ended up `VERSION_GREATER 3.11` which surely isn't the same thing. Mike's cmake was right in between, it would pass the greater 3.11 test while actually not support the `-j N` flag I was passing.

Generally speaking, I would suggest bumping the top level CMakeLists;txt `CMAKE_MINIMUM_REQUIRED(VERSION 2.8)` to something more sensible

https://github.com/NREL/EnergyPlus/blob/639da80b63a06653f6b121eb1698425511f3ef03/CMakeLists.txt#L10

**Review by mjwitte — APPROVED** [review]
With my shiny new cmake (v3.15.2 instead of 3.11.3) this all works now.
@Myoldmopar I'll let you make the final review and merge.

File: cmake/Install.cmake [hunk 1]
```diff
@@ -500,18 +500,25 @@ if ( BUILD_DOCS )
...
     if(NOT N EQUAL 0)
       set(DOC_BUILD_FLAGS "-j ${N}")
-      message("DOC_BUILD_FLAGS=${DOC_BUILD_FLAGS}")
     endif()
   endif()
-  install(CODE "execute_process(COMMAND \"${CMAKE_COMMAND}\" --build \"${CMAKE_BINARY_DIR}\" ${DOC_BUILD_FLAGS} --target documentation)")
+  if(WIN32)
+    # Win32 is multi config, so you must specify a config when calling cmake.
+    # Let's just use Release, it won't have any effect on LaTeX anyways.
+    set(DOC_CONFIG_FLAG "--config Release")
```
Comment (Myoldmopar) [inline cmake/Install.cmake:516]: :+1:

File: cmake/Install.cmake [hunk 2]
```diff
@@ -500,18 +500,25 @@ if ( BUILD_DOCS )
 
   # So instead, we just used the number of threads that are available. That's not ideal, since it ignores any "-j N" option passed by the user
   # But LaTeX should run quickly enough to not be a major inconvenience.
-  # There no need to do that for Ninja for eg, so only do it for Make
+  # There no need to do that for Ninja for eg, so only do it for Make and MSVC
 
-  # flag -j to cmake --build was added at 3.12
-  if(CMAKE_GENERATOR MATCHES "Make" AND (CMAKE_VERSION VERSION_GREATER "3.11"))
+  # flag -j to cmake --build was added at 3.12 (VERSION_GREATER_EQUAL need cmake >= 3.7, we apparently support 2.8...)
```
Comment (Myoldmopar) [inline cmake/Install.cmake:505]: Yeah, we will increment up the required CMake after this release and start cleaning out some of the bandaids.

File: cmake/Install.cmake [hunk 3]
```diff
@@ -443,7 +443,7 @@ elseif(WIN32)
   # You need at least one "install(..." command for it to be registered as a component
   install(CODE "MESSAGE(\"Registering filetypes.\")" COMPONENT RegisterFileType)
   install(CODE "MESSAGE(\"Copying and Registering DLLs\")" COMPONENT CopyAndRegisterSystemDLLs)
-
+  install(CODE "MESSAGE(\"Creating start menu.\")" COMPONENT CreateStartMenu)
```
Comment (Myoldmopar) [inline cmake/Install.cmake:446]: :+1:

File: cmake/qtifw/install_operations.qs [hunk 4]
```diff
@@ -27,18 +27,7 @@ function Component()
...
-      component.addOperation("CreateShortcut", "@TargetDir@/PreProcess/EPDraw/EPDrawGUI.exe", "@StartMenuDir@/EPDrawGUI.lnk");
-      component.addOperation("CreateShortcut", "@TargetDir@/EP-Launch.exe", "@StartMenuDir@/EP-Launch.lnk");
-      component.addOperation("CreateShortcut", "@TargetDir@/ExampleFiles/ExampleFiles.html", "@StartMenuDir@/Example Files Summary.lnk");
-      component.addOperation("CreateShortcut", "@TargetDir@/ExampleFiles/ExampleFiles-ObjectsLink.html", "@StartMenuDir@/ExampleFiles Link to Objects.lnk");
-      component.addOperation("CreateShortcut", "@TargetDir@/PreProcess/IDFEditor/IDFEditor.exe", "@StartMenuDir@/IDFEditor.lnk");
-      component.addOperation("CreateShortcut", "@TargetDir@/PreProcess/IDFVersionUpdater/IDFVersionUpdater.exe", "@StartMenuDir@/IDFVersionUpdater.lnk");
-      component.addOperation("CreateShortcut", "@TargetDir@/readme.html", "@StartMenuDir@/Readme Notes.lnk");
-      component.addOperation("CreateShortcut", "@TargetDir@/PreProcess/WeatherConverter/Weather.exe", "@StartMenuDir@/Weather Statistics and Conversions.lnk");
-
+      // Create Shortcuts in the Windows Start Menu: done separately (optional)
```
Comment (Myoldmopar) [inline cmake/qtifw/install_operations.qs:30]: I like this broken out anyway.

**Review by Myoldmopar — APPROVED** [review]

**Comment by Myoldmopar** [comment]
And for the record, the Windows CI box has CMake 3.15.0, so that won't be an issue there.  I'm merging this in and I'll kick off a package build tonight just to verify things are still good.

**Comment by jmarrec** [comment]
Good idea, I was going to suggest tagging it to check that CI can indeed make packages too. Thanks for the review.

**Pull request merged**